lib/core/test/unit/Cardano/Wallet/Gen.hs

```diff
@@ -222,6 +223,18 @@ genTxMetadata = do
         Left e -> fail $ show e <> ": " <> show (Aeson.encode json)
         Right metadata -> pure metadata
 
+-- | Generates a 'TxMetadata' containing only simple values.
+genSmallTxMetadata :: Gen TxMetadata
+genSmallTxMetadata = TxMetadata <$>
+    (Map.singleton <$> arbitrary <*> genSimpleTxMetadataValue)
```

Comment:
Hi @rvl 

Just curious: will having only one entry in the map affect the coverage in any way?

(In other words, do we have test conditions that only trigger in the presence of many-valued metadata maps?)

Reply:
Yes, thanks for the thoughts...

I thought it would be unlikely because the metadata is stored in the data as a json string.
That serialization is unit tested elsewhere.
Using a complex TxMetadata generator just makes things slower and less predictable.
A simple TxMetadata generator can still generate arbitrarily long strings with arbitrarily funny characters.